### PR TITLE
Add devcontainer and make some clean up

### LINKED ISSUE #30: Unnecessary dependencies
In `setup.cfg` the following dependencies are listed

```
ipython>=6
nbformat>=4
nbconvert>=5
requests>=2
```

Isn't that a mistake?

I also see `Js2Py` being used in `scxml.py` but only listed in `requirements_dev.txt`. That seems wrong as well.

Let me know if any of this is intended and then I'll fix in a PR.

### PR DESCRIPTION
**Pull request by mr-bjerre**
- [x] Add the option to use [VSCode Remote Containers](https://code.visualstudio.com/docs/remote/containers) for easy streamlined development environment.
- [x] Removed unused dependencies as well as some leftover files. Fixes #30 
- [ ] Moved `scxml.py` from `xstate` to `tests` since it's just test utility.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.177.0/containers/python-3/.devcontainer/base.Dockerfile
+
+# [Choice] Python version: 3, 3.9, 3.8, 3.7, 3.6
+ARG VARIANT="3.9"
+FROM mcr.microsoft.com/vscode/devcontainers/python:0-${VARIANT}
+
+# [Option] Install Node.js
+ARG INSTALL_NODE="true"
+ARG NODE_VERSION="lts/*"
+RUN if [ "${INSTALL_NODE}" = "true" ]; then su vscode -c "umask 0002 && . /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} 2>&1"; fi
```
Comment (davidkpiano) [inline .devcontainer/Dockerfile:10]: Do we need Node?

File: README.md [hunk 2]
```diff
@@ -31,11 +31,22 @@ More advanced examples in [the "examples" folder](./examples)
...
+1. Run `pytest` to run the tests! 👩‍🔬 (or run the `Run tests` task via VSCode)
+
+### Or on local drive
+
 1. Run `python3.7 -m venv .venv` to create a virtual environment
 1. Run `source .venv/bin/activate` to go into that virtual environment
 1. Run `pip install -r requirements_dev.txt` to install all of the dependencies in `requirements.txt` (which includes `pytest`)
 1. Make sure test files are present and up to date by running `git submodule init` (first time) and `git submodule update`
-1. Run `pytest` to run the tests! 👩‍🔬
+1. Run `pytest` to run the tests! 👩‍🔬 (or run the `Run tests` task via VSCode)
```
Comment (davidkpiano) [inline README.md:49]:
```suggestion
1. Run `pytest` to run the tests! 👩‍🔬 (or run the `Run tests` task via VS Code)
```

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (mr-bjerre) [reply .devcontainer/Dockerfile:10]: Ah no that was before you had the `npm` call

File: tests/test_scxml.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 from typing import Optional, Dict, List
 import json
 import pytest
-from xstate.scxml import scxml_to_machine
+from tests.scxml import scxml_to_machine
```
Comment (davidkpiano) [inline tests/test_scxml.py:5]: We might want `scxml_to_machine` to be public API

Comment (mr-bjerre) [reply tests/test_scxml.py:5]: Ah yes why not. I'll move it back tomorrow

**Comment by mr-bjerre** [comment]
Should be ready for merge now

**Pull request merged**